### PR TITLE
fixes #59, baseurl to resolve // issue

### LINKED ISSUE #59: Having a baseurl of '' doesnt work
An empty base url is recommended for jekyll hosted at the root of the site but this causes most of the links to break. 
Using '/' makes things mostly work, but there are a few places which result in a // inside the URL (such as inside the feed.xml, and when doing things with plugins like `jekyll-redirect-from`.

This article has some really good instructions on how to do baseurl stuff.
http://downtothewire.io/2015/08/15/configuring-jekyll-for-user-and-project-github-pages/

### PR DESCRIPTION
**Pull request by lukemcgregor**
Applied the ruleset for baseurls outlined here

http://downtothewire.io/2015/08/15/configuring-jekyll-for-user-and-project-github-pages/

### CODE REVIEW

**Comment by lukemcgregor** [comment]
Had to open new pull as last one was completely broken

**Comment by joshgerdes** [comment]
@lukemcgregor Sorry for the delay, I finally got around to reviewing these PRs and merging them. Thank you for your contributions. When you get a chance can you update this PR so we can get it in.  Looks like the fix will make life easier for a lot of people. Cheers!

**Comment by lukemcgregor** [comment]
@joshgerdes try now. Thanks for making this awesome skin 👍 

If you are interested heres where I'm using it: http://blog.staticvoid.co.nz/2017/migrating-my-blog-to-jekyll-and-github-pages/

**Review by joshgerdes — APPROVED** [review]

File: _includes/head.html [hunk 1]
```diff
@@ -28,27 +28,27 @@
...
+  <link rel="manifest" href="{{ site.baseurl }}/images/favicons/manifest.json">
+  <link rel="shortcut icon" href="{{ site.baseurl }}/images/favicons/favicon.ico">
   <meta name="msapplication-TileColor" content="#ffc40d">
-  <meta name="msapplication-TileImage" content="{{ "images/favicons/mstile-144x144.png" | prepend: site.baseurl }}">
+  <meta name="msapplication-TileImage" content="{{ site.baseurl }}/images/favicons/mstile-144x144.png">
   <meta name="theme-color" content="#ffffff">
-
-  <link rel="stylesheet" href="{{ "css/main.css" | prepend: site.baseurl }}?{{site.time | date: '%s%N'}}">
+  
+  <link rel="stylesheet" href="{{ site.baseurl }}/css/main.css?{{site.time | date: '%s%N'}}">
```
Comment (adishavit) [inline _includes/head.html:51]: @lukemcgregor : this line looks for `css/main.css` which does not exist after this commit.

**Pull request merged**